jaxrs-client/src/main/java/io/micronaut/jaxrs/client/JaxRsConfiguration.java

```diff
@@ -0,0 +1,510 @@
+
+/**
+ * The JAX-RS Client configuration.
+ *
+ * @author Denis Stepanov
+ * @since 4.6
+ */
+@Internal
+final class JaxRsConfiguration implements Configuration {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JaxRsConfiguration.class);
+
+    private final Map<String, Object> properties;
+    private final List<Component> components;
+
+    private List<ReaderInterceptor> readerInterceptors;
+    private List<WriterInterceptor> writerInterceptors;
+    private List<JaxRsMessageBodyReaderDefinition> readers;
+    private List<JaxRsMessageBodyWriterDefinition> writers;
+    private List<ClientRequestFilter> requestFilters;
+    private List<ClientResponseFilter> responseFilters;
+
+    public JaxRsConfiguration() {
+        this(new LinkedHashMap<>(), new ArrayList<>());
+    }
+
+    public JaxRsConfiguration(Map<String, Object> properties, List<Component> components) {
+        this.properties = properties;
+        this.components = components;
+    }
+
+    JaxRsConfiguration copy() {
+        return new JaxRsConfiguration(new LinkedHashMap<>(properties), new ArrayList<>(components));
+    }
+
+    public void addProperty(String name, Object value) {
+        properties.put(name, value);
+    }
+
+    void register(Class<?> componentClass) {
+        register(componentClass, 0);
+    }
+
+    void register(Class<?> componentClass, int priority) {
+        register(componentClass, priority, new Class<?>[0]);
+    }
+
+    void register(Class<?> componentClass, Class<?>... contracts) {
+        if (contracts == null || contracts.length == 0) {
+            return;
+        }
+        register(componentClass, 0, contracts);
+    }
+
+    private List<ComponentContract> toContracts(Class<?>[] contracts) {
+        return Arrays.stream(contracts).map(c -> new ComponentContract(c, 0)).toList();
+    }
+
+    void register(Class<?> componentClass, int priority, Class<?>... contracts) {
+        components.add(new ClassComponent(componentClass, priority, toContracts(contracts)));
+    }
+
+    void register(Class<?> componentClass, Map<Class<?>, Integer> contracts) {
+        components.add(new ClassComponent(componentClass, 0, toContracts(contracts)));
+    }
+
+    private List<ComponentContract> toContracts(Map<Class<?>, Integer> contracts) {
+        return contracts.entrySet().stream().map(e -> new ComponentContract(e.getKey(), e.getValue())).toList();
+    }
+
+    void register(Object component) {
+        register(component, 0);
+    }
+
+    void register(Object component, int priority) {
+        register(component, priority, new Class<?>[0]);
+    }
+
+    void register(Object component, Class<?>... contracts) {
+        if (contracts == null || contracts.length == 0) {
+            return;
+        }
+        register(component, 0, contracts);
+    }
+
+    void register(Object component, int priority, Class<?>... contracts) {
+        components.add(new InstanceComponent(component, priority, toContracts(contracts)));
```

Comment:
maybe invalidate the readers/writers fields? Or is this not necessary

Reply:
I think it's fine for now